Source/engine/animationinfo.h

```diff
@@ -37,41 +37,46 @@ enum AnimationDistributionFlags : uint8_t {
  */
 class AnimationInfo {
 public:
-	/**
-	 * @brief Animation sprite
-	 */
-	std::optional<CelSprite> celSprite;
-	/**
-	 * @brief How many game ticks are needed to advance one Animation Frame
-	 */
-	int TicksPerFrame;
-	/**
-	 * @brief Increases by one each game tick, counting how close we are to TicksPerFrame
-	 */
-	int TickCounterOfCurrentFrame;
-	/**
-	 * @brief Number of frames in current animation
-	 */
-	int NumberOfFrames;
-	/**
-	 * @brief Current frame of animation
-	 */
-	int CurrentFrame;
-	/**
-	 * @brief Is the animation currently petrified and shouldn't advance with gfProgressToNextGameTick
-	 */
-	bool IsPetrified;
+	// clang-format off
+	// generic getters
+	[[nodiscard]] const std::optional<CelSprite>  &getCelSprite() const { return celSprite_; }
+	[[nodiscard]] int getCurrentFrame()                           const { return currentFrame_; }
+	[[nodiscard]] int getNumberOfFrames()                         const { return numberOfFrames_; }
+	[[nodiscard]] int getTickCounterOfCurrentFrame()              const { return tickCounterOfCurrentFrame_; }
+	[[nodiscard]] int getTicksPerFrame()                          const { return ticksPerFrame_; }
+	[[nodiscard]] int getTriggerFrame()                           const { return triggerFrame_; }
+
+	// helper getters
+	[[nodiscard]] bool isLastFrame()         const { return (currentFrame_ == numberOfFrames_ - 1); }
+	[[nodiscard]] bool isOver()              const { return (currentFrame_ >= numberOfFrames_ - 1); }

```

Comment:
I think both could be combined to one function.
Cause `currentFrame` should never be more then `numberOfFrames` else it's a bug.

Another note:
`isOver` is not 100% correct.
If you have a animation with 5 `NumberOfFrames` and each frame has a `TicksPerFrame` of 10.
Then if you have `currentFrame`=5 and `tickCounterOfCurrentFrame` = 0. The animation is only over after 9 more game ticks. 😉 
It's not a problem in the current codebase, cause on the locations where this check is performed all animations have `TicksPerFrame` = 1.
A solution would be to also check for `tickCounterOfCurrentFrame`, cause it shouldn't matter.
But better check twice, it was a little while away when I last checked all animations. 😁

Reply:
:shrug: what to do. Plenty of this condition in the code right now...
`currentFrame_ >= numberOfFrames_ - 1`
thus this helper. I think since there is distinction like that - I just maintained it. It doesn't change current behavior, nor make it better, than just add some readability.
I can imagine there are some cases where it exactly have to be equal. Changing this to equal or greater could maybe introduce some bugs, or some events occur more often, so I didn't want to touch the logic behind.

Reply:
> I can imagine there are some cases where it exactly have to be equal.
I think we are pretty safe to change this. 🙂 
All occurrences that change `currentFrame` don't set it to something great then `numberOfFrames`.
But perhaps I miss something.

> Changing this to equal or greater could maybe introduce some bugs, or some events occur more often, so I didn't want to touch the logic behind.
I'm more concerned about `triggerFrame`, cause it needs to be updated every time we change the mode. 😉
But this is another discussions.